src/components/Menu/index.tsx

```diff
@@ -120,10 +120,9 @@ export function Item({children, label, style, onPress, ...rest}: ItemProps) {
       onFocus={onFocus}
       onBlur={onBlur}
       onPress={async e => {
-        await onPress(e)
-        if (!e.defaultPrevented) {
-          control?.close()
-        }
+        control?.close(() => {
```

Comment:
@haileyok What do you think about creating a custom hook, `useMenuControlContext`, instead of using `React.useContext` directly? This way, we could ensure that the control context for the Menu is always defined and never undefined.

```js
export const useMenuControlContext = () => {
  const context = React.useContext(Context)

  if (!context) {
    throw new Error('useMenuControlContext must be used within a Context.Provider')
  }

  return context
}
```
Also, this would remove this `@ts-ignore`

![Screenshot 2024-11-12 at 14 12 46](https://github.com/user-attachments/assets/5a8d36d0-f2e3-4399-bd0f-879f31d4203c)


`control?` is no longer needed in line 123
![Screenshot 2024-11-12 at 14 13 57](https://github.com/user-attachments/assets/5c0ae3fb-9ef5-4cf3-bbca-4bcaa0ef575e)

Notes: 
* It exists [this function](https://github.com/bluesky-social/social-app/blob/main/src/components/Menu/index.tsx#L28-L30) that is supposed to be this custom hook I'm mentioning, but there's a typo in there `Memo`, and it is not handling undefined value. 
* The Menu/index.web.tsx file also have this unused custom hook, and in there's a [duplication of the context hook ](https://github.com/bluesky-social/social-app/blob/main/src/components/Menu/index.web.tsx#L45-L47), that's why I think it's better to export it from the `context.tsx`, to avoid unnecessary duplication.
* This custom hook is suggested by Kent C Dodds in his [blog post](https://kentcdodds.com/blog/how-to-use-react-context-effectively#conclusion)

Reply:
Oop, yea need to make sure this PR doesn't get lost down in my stack... @estrattonbailey any thoughts on the suggested approach here? Think you have the most context on what we need out of this.

Reply:
Yep this is a good suggestion, knocked that out. Appreciate the thoughts!